### PR TITLE
[coq] bump Coq and Dune

### PR DESCRIPTION
**Pull request by ejgallego**
- [x] Abel
  - [x] :chart_with_upwards_trend:  updated 
- [x] bigenough
  - [x] :chart_with_upwards_trend:  updated 
- [x] CompCert
  - [x] :chart_with_upwards_trend:  updated 
- [x] Coq-Equations
  - [x] :chart_with_upwards_trend:  updated 
- [x] coq-master
  - [x] :chart_with_upwards_trend:  updated 
- [x] dune-ocaml
  - [x] :chart_with_upwards_trend:  updated 
- [x] finmap
  - [x] :chart_with_upwards_trend:  updated 
- [x] flocq
  - [x] :chart_with_upwards_trend:  updated 
- [x] fourcolor
  - [x] :chart_with_upwards_trend:  updated 
- [x] gaia
  - [x] :chart_with_upwards_trend:  updated 
- [x] Coq-HoTT
  - [x] :chart_with_upwards_trend:  updated 
- [x] hydra-battles
  - [x] :chart_with_upwards_trend:  updated 
- [x] LibHyps
  - [x] :chart_with_upwards_trend:  updated 
- [x] math-comp
  - [x] :chart_with_upwards_trend:  updated 
- [x] mczify
  - [x] :chart_with_upwards_trend:  updated 
- [x] multinomials
  - [x] :chart_with_upwards_trend:  updated 
- [x] odd-order
  - [x] :chart_with_upwards_trend:  updated 
- [x] paramcoq
  - [x] :chart_with_upwards_trend:  updated 
- [x] real-closed
  - [x] :chart_with_upwards_trend:  updated 
- [x] tarjan
  - [x] :chart_with_upwards_trend:  updated

### CODE REVIEW

**Review by Alizter — APPROVED** [review]

**Comment by ejgallego** [comment]
Umm, actually I broke Dune boot composition in https://github.com/ocaml/dune/pull/5866 :S

**Comment by ejgallego** [comment]
I bumped a bunch of deps, but still we get some errors, in particular it seems like stuff that is outside Coq's CI?

**Comment by ejgallego** [comment]
Note that the Coq tree has 3 patches:
- Disable Print and About output (as to focus on the warnings)
- Disable the META output by coqdep (triggers the bug in dune)
- Reenable the `theories/dune` file

Only the last patch is desired, the rest should be fixed upstream.

**Comment by Alizter** [comment]
@ejgallego Can you rebase on main again, I fixed and merged the nix patch.

**Comment by ejgallego** [comment]
Rebase done, great that we got nix working.

**Comment by Alizter** [comment]
> will it cache the CI too?

It will only cache its immediate setup, but the things we do in opam such as installing zarith/menhir and the build are not cached.

**Comment by ejgallego** [comment]
> > will it cache the CI too?
> 
> It will only cache its immediate setup, but the things we do in opam such as installing zarith/menhir and the build are not cached.

It'd be nice if we could implement something like the dune cache in the other setup.

**Comment by Alizter** [comment]
> It'd be nice if we could implement something like the dune cache in the other setup.

I think it should be possible. Just have to tar and cache the dune cache between the workflows. The OCaml action already does that, so perhaps I could find a way to do it.

**Comment by ejgallego** [comment]
> I think it should be possible. Just have to tar and cache the dune cache between the workflows. The OCaml action already does that, so perhaps I could find a way to do it.

Is there something like a Nix OCaml action?

**Comment by Alizter** [comment]
@ejgallego Can you rebase and push your local branch? I've messed this one up.

**Comment by ejgallego** [comment]
Done

**Comment by ejgallego** [comment]
Note that the idea of this PR was to bump only the minimum amount of submodules to make the CI work.

**Comment by Alizter** [comment]
@ejgallego yes I gathered. I made the list to keep track.

File: .gitmodules [hunk 1]
```diff
@@ -8,7 +8,7 @@
 	branch = dune
 [submodule "coq-master"]
 	path = coq-master
-	url = https://github.com/coq/coq.git
+	url = https://github.com/ejgallego/coq.git
 	branch = master
```
Comment (Alizter) [inline .gitmodules:12]: The submodule patch doesn't exist in master, you probably want to push your changes to a branch and point there.

**Comment by Alizter** [comment]
Many projects no longer work and instead require an explicit plugins field. Some change in coqdep has caused thi.

**Comment by ejgallego** [comment]
> Many projects no longer work and instead require an explicit plugins field. Some change in coqdep has caused this.

That's very weird, I don't observe that.

File: .gitmodules (hunk 1 above)
Comment (ejgallego) [reply .gitmodules:12]: I see the submodule commit just fine, as does the CI

**Comment by Alizter** [comment]
@ejgallego I don't have a global Coq install.

**Comment by ejgallego** [comment]
> @ejgallego I don't have a global Coq install.

What is a global Coq install?

Neither the CI or my dev setup have Coq installed, if that's what you mean.

Can you show me the error you are talking about in the CI log?

**Comment by Alizter** [comment]
The error is the same one as always from Env: Blah blah coqlib not found something something `-noinit -boot`. The problem is, so many things can trigger this.

**Comment by ejgallego** [comment]
> The error is the same one as always from Env: Blah blah coqlib not found something something `-noinit -boot`.

What is the line in the CI log?

> The problem is, so many things can trigger this.

What are these things? AFAICT there is only one condition that can make coqlib not found.

**Comment by Alizter** [comment]
I must be doing something wrong. I don't have time to start up coq-universe right now unfortunately.

**Comment by ejgallego** [comment]
If you can reproduce just sent the report logs along with git status --recursive output

**Comment by ejgallego** [comment]
@Alizter this is up to date for Coq and Dune, submodules need more bumps, unfortunately.

**Comment by Alizter** [comment]
Hydra battles now depends on bignums and coqprime for which I have added preliminary dune support.

**Comment by Alizter** [comment]
I've updated all the submodules by hand, hopefully the CI agrees with my remedy.

**Comment by Alizter** [comment]
Merry Christmas the CI is green.

**Comment by ejgallego** [comment]
> Merry Christmas the CI is green.

ho ho ho !!! Great!!!

**Pull request merged**